kcp/utils.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kcp
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/go-logr/logr"
+	kcpv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func KcpAPIsGroupPresent(restConfig *rest.Config, setupLog logr.Logger) bool {
+	discoveryClient, err := discovery.NewDiscoveryClientForConfig(restConfig)
+	if err != nil {
+		setupLog.Error(err, "failed to create discovery client")
+		os.Exit(1)
+	}
+	apiGroupList, err := discoveryClient.ServerGroups()
+	if err != nil {
+		setupLog.Error(err, "failed to get server groups")
+		os.Exit(1)
+	}
+
+	for _, group := range apiGroupList.Groups {
+		if group.Name == kcpv1alpha1.SchemeGroupVersion.Group {
+			for _, version := range group.Versions {
+				if version.Version == kcpv1alpha1.SchemeGroupVersion.Version {
+					return true
+				}
+			}
+		}
+	}
+	return false
+}
+
+// GetRestConfigForAPIExport returns a *rest.Config properly configured to communicate with the endpoint for the
+// APIExport's virtual workspace.
+func GetRestConfigForAPIExport(ctx context.Context, cfg *rest.Config, apiExportName string, setupLog logr.Logger) (*rest.Config, error) {
```

Comment:
Camelcase guidelines.
```suggestion
func GetRestConfigForApiExport(ctx context.Context, cfg *rest.Config, apiExportName string, setupLog logr.Logger) (*rest.Config, error) {
```